src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionService.java

```diff
@@ -47,23 +55,46 @@ public Subscription createSubscription(Subscription subscription, String actioni
         return repository.save(subscription);
     }
 
-    public void configureListTypesForSubscription(String userId, List<String> listType) {
-        log.info(writeLog(userId, UserActions.CREATE_SUBSCRIPTION, LOCATION_ID.name()));
+    public void addListTypesForSubscription(SubscriptionListType subscriptionListType,
+                                                  String actioningUserId) {
+        log.info(writeLog(actioningUserId, UserActions.CREATE_SUBSCRIPTION, LOCATION_ID.name()));
+        //DELETE EXISTING LIST TYPE CONFIG FOR A USER
+        duplicateListTypeHandler(subscriptionListType);
+        subscriptionListTypeRepository.save(subscriptionListType);
+    }
+
+    public void configureListTypesForSubscription(SubscriptionListType subscriptionListType,
+                                                  String actioningUserId) {
+        log.info(writeLog(actioningUserId, UserActions.CREATE_SUBSCRIPTION, LOCATION_ID.name()));
 
-        repository.updateLocationSubscriptions(userId,
-            listType == null ? "" : StringUtils.join(listType, ','));
+        subscriptionListTypeRepository.updateLocationSubscriptions(subscriptionListType.getUserId(),
+            subscriptionListType.getListType() == null ? "" :
+                StringUtils.join(subscriptionListType.getListType(), ','));
     }
 
     public void deleteById(UUID id, String actioningUserId) {
         Optional<Subscription> subscription = repository.findById(id);
-
         if (subscription.isEmpty()) {
             throw new SubscriptionNotFoundException(String.format(
                 "No subscription found with the subscription id %s",
                 id
             ));
         }
 
+        int minNoOfSubscriptions = 1;
+        if (subscription.get().getSearchType().equals(LOCATION_ID)) {
+            Optional<SubscriptionListType> subscriptionListTypes =
+                subscriptionListTypeRepository.findByUserId(subscription.get().getUserId());
+            List<Subscription> userSubscriptions = repository.findLocationSubscriptionsByUserId(
+                subscription.get().getUserId());
+            //FIND ALL THE LOCATION SUBSCRIPTION FOR THE USER AND CHECK IF MORE THAN ONE LOCATION SUBSCRIPTION EXISTS
+            //DO NOT DELETE RECORD FROM SUBSCRIPTION LIST TYPE BECAUSE ONE RECORD IS LINKED WITH ALL THE LOCATION
+            //SUBSCRIPTIONS
+            if (userSubscriptions.size() <= minNoOfSubscriptions) {
```

Comment:
The logic here is quite similar to the one below and also the one in SubscriptionLocationService.

I wonder if this can be pulled out in some way so it's common?

Reply:
done.